compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1064,37 +1064,37 @@ class TreeUnpickler(reader: TastyReader,
           selfInfo = if (self.isEmpty) NoType else self.tpt.tpe
         ).integrateOpaqueMembers
 
-      val (constr, stats0) =
+      val constr =
         if nextByte == SPLITCLAUSE then
           assert(unpicklingJava, s"unexpected SPLITCLAUSE at $start")
           val tag = readByte()
           def ta = ctx.typeAssigner
           val flags = Flags.JavaDefined | Flags.PrivateLocal | Flags.Invisible
-          val pflags = Flags.JavaDefined | Flags.Param
-          val tdefRefs = tparams.map(_.symbol.asType)
           val ctorCompleter = new LazyType {
             def complete(denot: SymDenotation)(using Context) =
               val sym = denot.symbol
-              lazy val tparamSyms: List[TypeSymbol] = tparams.map: tdef =>
+              val pflags = flags | Flags.Param
+              val tparamRefs = tparams.map(_.symbol.asType)
+              lazy val derivedTparamSyms: List[TypeSymbol] = tparams.map: tdef =>
                 val completer = new LazyType {
                   def complete(denot: SymDenotation)(using Context) =
-                    denot.info = tdef.symbol.asType.info.subst(tdefRefs, tparamSyms.map(_.typeRef))
+                    denot.info = tdef.symbol.asType.info.subst(tparamRefs, derivedTparamRefs)
                 }
-                newSymbol(sym, tdef.name, pflags, completer, coord = cls.coord)
-              val paramSym =
+                newSymbol(sym, tdef.name, Flags.JavaDefined | Flags.Param, completer, coord = cls.coord)
+              lazy val derivedTparamRefs: List[Type] = derivedTparamSyms.map(_.typeRef)
```

Comment:
This is only used once. It can be a `def`.

Reply:
Also, maybe inline it back into `def complete`, like it was before?

Reply:
The point is that each type parameter has a new completer, so this is referenced arbitrary times